llvm/test/Transforms/GlobalOpt/cleanup-pointer-root-users.ll

```diff
@@ -12,7 +12,7 @@ define void @test1a() {
 
 define void @test1b(ptr %p) {
 ; CHECK-LABEL: @test1b(
-; CHECK-NEXT: store
+; CHECK-NOT: store
```

Comment:
This change looks incorrect to me: `%p` might be a malloc-ed pointer here, which would leak after the transform.

Reply:
So, in general, store instructions with a pointer as its value operand should not be removed?

Reply:
Not quite. The point of CleanupPointerRootUsers() is to remove assignments for which we know that the value is not dynamically allocated. So e.g. a store of a constant is fine, but a store of an unknown argument is not, because that argument could have come from malloc.